### PR TITLE
Add self-directed help section to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jtigger**
The intent of this pull request is to get early feedback on the general expression of the idea sketched in exercism/discussions#73.  _It is not intended to be merged in its current form._

You can view it here: https://github.com/jtigger/exercism.io/blob/triage-contribution/CONTRIBUTING.md

You'll get the most complete feel following this path:
1. I see a problem...
2. ... with a specific exercise.

The general intent is for someone with not a lot of experience of behind-the-scenes exercism to quickly find the place and information needed to start contributing... whether it's "just" reporting an issue, submitting a fix/enhancement, or even taking on thinking about more strategic things.

This work will be followed by an effort to split this page and related pages into "modules" of units-of-work that will be referred from this "decision tree." (think: a set of `.md` files under `docs`, one per workflow).

Brutally honest feedback, please! :)

### CODE REVIEW

**Comment by kytrinyx** [comment]
Holy smokes I really like where this is going. The ability to drill down really quickly is powerful.

I'm wishing I had more brutally useful feedback, but I don't. I think you're on to something here.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -131,6 +142,8 @@ Are you a language enthusiast?  Help other people fall in love with your languag
 
 ### ... Ruby on Rails
 
+
+
```
Comment (kytrinyx) [inline CONTRIBUTING.md]: FYI we don't actually have any Rails in the project.

**Comment by Insti** [comment]
I also like what you're doing here.
(But don't have any specific feedback, just keep doing what you're doing?)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,6 +4,175 @@ First of all, **thank you** for helping with Exercism.io!
...
+### ... with the command-line client (CLI)
+
+* If you're having trouble installing or working with the CLI, ask for help in our [online support chat](https://gitter.im/exercism/support).
+- If the CLI has a bug, you can help by [filing a bug report](docs/filing-a-bug-report.md) in the [`exercism/cli` repository](https://github.com/exercism/cli/issues/new).
+  - After reporting the problem, if you have the time and the desire, you can help *even more* by [fixing it](#-go).
+
+-
+
+### ... getting started with a language
```
Comment (tejasbubane) [inline CONTRIBUTING.md]: These `...` seem a little odd and distracting to me. Can we have full sentences like `I see a problem getting started with a language`?
But this is just my opinion.

Something like this:

exercism-contrib

Comment (tejasbubane) [inline CONTRIBUTING.md]: > the headers of each landing section is a complete sentence and there are not ellipsis. Yes?

Yes. Looks better IMO. Lets see what others have to say.

Comment (jtigger) [inline CONTRIBUTING.md]: Ditto.  Thanks for the feedback!!  If you see think of anything else, big or small, please share!

**Comment by kytrinyx** [comment]
This is a tangent, and you've likely not even considered the aesthetics of it yet, but I've seen people use octicons and/or fontawesome to good effect.

https://octicons.github.com

- bug: https://octicons.github.com/icon/bug/
- idea: https://octicons.github.com/icon/light-bulb/
- help (?)
  - https://octicons.github.com/icon/gift/
  - https://octicons.github.com/icon/heart/
- website: https://octicons.github.com/icon/browser/
- exercise (?)
  - https://octicons.github.com/icon/keyboard/
  - https://octicons.github.com/icon/file-binary/
- ... etc

**Comment by jtigger** [comment]
Note, @kytrinyx, I've rebased over your branch to include references to the Contributor guides you've been working on (#3232) — great stuff!

**Comment by jtigger** [comment]
You're right: I have specifically avoided investing in the graphics part.  I'm the **LAST* guy you want picking graphics.  I grabbed some from Google image search that said they were "royalty free".

Re:octicons — very cool!

**Comment by jtigger** [comment]
At this point, I believe I've reached a milestone with this work and ready to submit for review.

@kytrinyx, I'm specifically deferring doing anything with graphics.  I would ❤️  to see someone with graphics design chops give these an Exercism vibe.  Far more important seemed to be the contextualizing of someone's intent.

I also followed your advice, @kytrinyx, and created issues for other sections that this guide links to.  Those pages point back to their issues, respectively.

Some leaves on this decision tree are stronger than others.  I hope by calling this a first iteration of "done", others can start to elaborate/tweak/adjust.

**Comment by jtigger** [comment]
This is such a significant chunk of work (with sub-parts pending), I'd love to either push this through or course-correct as needed.

@kytrinyx?

**Comment by kytrinyx** [comment]
@jtigger sorry, I've been traveling non-stop for the past 6 weeks. I'm still traveling, but I'm spending Thanksgiving reviewing PRs and cleaning things up in Exercism, so I'll take a look at this today or tomorrow.

**Comment by kytrinyx** [comment]
@jtigger this is an awesome first pass.

I saw an `its'` vs `it's` but I'm going to let that get fixed in some subsequent pass. Same with the images.

Thank you so much for taking the time to dig so deeply into this.

**Comment by ErikSchierboom** [comment]
Wonderful improvement! Great work @jtigger

**Comment by jtigger** [comment]
@kytrinyx I hope I cane across as a "squeeky wheel" and not impatient.  I'm delighted to get through a first iteration.  It's quite imperfect, but I have hope in this approach.

We could really use a designer's eye on this:

- what are the common goals, workflows and attendant ux?
- better graphics!!! and
perhaps ideals on making the visual design easier on the eyes?

**Comment by kytrinyx** [comment]
We sure could use a designer's eye!

It feels like there are some different categories of contribution docs:

1. Exercism, the Product
1. Exercism, the Curriculum and Language Tracks
1. GitHub / Workflow / Conventions

And within each of those categories, there are lots of sub-categories.

The thing that is so hard for me to figure out is—where are people starting from? Where are they when they decide to contribute? What do they find first? I have literally not a single clue.

**Comment by jtigger** [comment]
@kytrinyx 
> The thing that is so hard for me to figure out is—where are people starting from? Where are they when they decide to contribute? What do they find first? I have literally not a single clue.

EXACTLY!  That has been one of the most challenging parts in doing this.  I spent most of my time on the PR sitting there trying to imagine what are those inbound paths.

It's really easy to want to speak to our audience in terms that *we* understand... when it's an order-of-magnitude better experience when we shoulder up with them and speak from their perspective.

One thing that started to become clear to me is that this guide is aiming to help the uninitiated.... once someone starts to wrap their arms around the organization and culture of Exercism, such a guided tour becomes less and less useful — there are richer sources of help in terms of the relationships this person has established.  So, the good news is that we don't have to account for *all* paths, just the common ones that new-to-exercism folks might hit.

I've wanted to contribute when:

- I go through the on boarding process and hit snags.
- I've seen a bug in the website or in the CLI.
- I'm reviewing others' submissions and start wishing for features that optimizes my workflow.
- I'm solving an exercise and find problems with it (usually I'm frustrated with the API spelled out by the tests... but also by the value/path the tests encourage... and then other times the language of the `description.md`).
- ...

**Pull request merged**